specs/experimental/contracts/legacy/resolved-delegate-proxy.md

```diff
@@ -0,0 +1,119 @@
+# ResolvedDelegateProxy
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [Definitions](#definitions)
+- [Assumptions](#assumptions)
+  - [a01-001: AddressManager provides valid implementation addresses](#a01-001-addressmanager-provides-valid-implementation-addresses)
+    - [Mitigations](#mitigations)
+  - [a01-002: Implementation storage layout compatibility](#a01-002-implementation-storage-layout-compatibility)
+    - [Mitigations](#mitigations-1)
+- [Invariants](#invariants)
+  - [i01-001: Implementation address resolution requirement](#i01-001-implementation-address-resolution-requirement)
+    - [Impact](#impact)
+  - [i01-002: Immutable configuration after deployment](#i01-002-immutable-configuration-after-deployment)
+    - [Impact](#impact-1)
+- [Function Specification](#function-specification)
+  - [constructor](#constructor)
+  - [fallback](#fallback)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+The ResolvedDelegateProxy is a legacy proxy contract that dynamically resolves its implementation address through the
+[AddressManager](./address-manager.md) contract. It maintains backwards compatibility with pre-Bedrock versions of the
+Optimism system where proxy implementations were managed through a centralized name registry rather than standard proxy
+patterns.
+
+## Definitions
+
+N/A
+
+## Assumptions
+
+### a01-001: AddressManager provides valid implementation addresses
+
+The [AddressManager](./address-manager.md) contract returns valid, non-zero implementation addresses when queried with
+the configured implementation name. If the AddressManager returns the zero address or an invalid address, the proxy will
+fail to function.
+
+#### Mitigations
+
+- The fallback function explicitly checks for zero address and reverts with a clear error message
+- AddressManager is controlled by governance with established processes for address updates
+- Legacy contracts using this proxy pattern are being phased out in favor of standard proxy patterns
+
+### a01-002: Implementation storage layout compatibility
+
+The implementation contract does not use mappings in storage slots 0 or 1, which would conflict with the proxy's
+`implementationName` and `addressManager` mappings.
+
+#### Mitigations
+
+- This storage pattern was a known design constraint in the legacy system
+- Implementation contracts were designed with awareness of these reserved slots
+- Modern proxy patterns (used in current OP Stack) avoid this issue entirely
+
+## Invariants
```

Comment:
devinai Missing invariant, please, this contract obviously needs to behave like a transparent proxy for end-users